File: plugins/kraken_bundle/android/src/main/java/com/taobao/kraken_bundle/BundleManager.java
package com.taobao.kraken_bundle;

public class BundleManager {

  private static BundleManager instance;

  private String bundleUrl;
  private String zipBundleUrl;
  private KrakenBundlePlugin krakenBundlePlugin;


  private BundleManager() {
  }


  public static BundleManager getInstance() {
    if (instance == null) {
      synchronized (BundleManager.class) {
        if (instance == null) {
          instance = new BundleManager();
        }
      }
    }
    return instance;
  }

  /**
   * page load form
   * priority bundleUrl > zipBundleUrl
   * @param bundleUrl
   * @param zipBundleUrl
   */
  public void setUp(String bundleUrl, String zipBundleUrl) {
    this.bundleUrl = bundleUrl;
    this.zipBundleUrl = zipBundleUrl;
  }

  public String getBundleUrl() {
    return "https://dev.g.alicdn.com/kraken/kraken-demos/long-list/build/kraken/index.js";
  }

  public String getZipBundleUrl() {
    return zipBundleUrl;
  }

  public void reload() {
    if (krakenBundlePlugin != null) {
      krakenBundlePlugin.reload();
    }
  }

  public void onAttach(KrakenBundlePlugin krakenBundlePlugin) {
    this.krakenBundlePlugin = krakenBundlePlugin;
  }

  public void onDetach() {
    this.krakenBundlePlugin = null;
  }

  public void destory() {
    instance = null;
  }
}

File: plugins/kraken_bundle/macos/Classes/BundleManager.swift
//
//  BundleManager.swift
//  kraken_bundle
//
//  Created by lzl on 2020/3/27.
//

import Cocoa
import FlutterMacOS

public class BundleManager: NSObject {
  private static let instance:BundleManager = BundleManager()
  private var bundleUrl:String?
  private var zipBundleUrl:String?
  private weak var krakenBundlePlugin:KrakenBundlePlugin?
  private weak var channel:FlutterMethodChannel?
  
  private override init() {
    bundleUrl = nil
    zipBundleUrl = nil
  }
  
  public static var shared: BundleManager {
      return self.instance
  }
  /**
   * page load form
   * priority bundleUrl > zipBundleUrl
   * @param bundleUrl
   * @param zipBundleUrl
   */
  public func setUp(bundleUrl:String, zipBundleUrl:String) {
    self.bundleUrl = bundleUrl;
    self.zipBundleUrl = zipBundleUrl;
  }

  public func getBundleUrl()->String? {
    return bundleUrl
  }

  public func getZipBundleUrl()->String? {
    return zipBundleUrl;
  }

  public func reload() {
    if (self.krakenBundlePlugin != nil) {
      self.channel?.invokeMethod("reload", arguments: nil)
    }
  }

  public func onAttach(krakenBundlePlugin:KrakenBundlePlugin, channel:FlutterMethodChannel) {
    self.krakenBundlePlugin = krakenBundlePlugin
    self.channel = channel
  }
}
